tmt/steps/prepare/artifact/providers/koji.py

```diff
@@ -103,29 +110,67 @@ class KojiArtifactProvider(ArtifactProvider[RpmArtifactInfo]):
         artifacts = provider.download_artifacts(guest, Path("/tmp"), [])
     """
 
-    # TODO: Make RPM_BASE_URL configurable via FMF/CLI, not just env var
-    API_URL = os.getenv("KOJI_API_URL", "https://koji.fedoraproject.org/kojihub")  # For metadata
+    SUPPORTED_PREFIXES = ("koji.build:", "koji.task:", "koji.nvr:")
+
+    def __new__(cls, raw_provider_id: str, logger: tmt.log.Logger) -> 'KojiArtifactProvider':
+        """
+        Create a specific Koji provider based on the ``raw_provider_id`` prefix.
+
+        The supported provides are:
+        :py:class:`KojiBuild`,
+        :py:class:`KojiTask`,
+        :py:class:`KojiNvr`.
+
+        :raises ValueError: If the prefix is not supported
+        """
+        if raw_provider_id.startswith("koji.build:"):
+            return super().__new__(KojiBuild)
+        if raw_provider_id.startswith("koji.task:"):
+            return super().__new__(KojiTask)
+        if raw_provider_id.startswith("koji.nvr:"):
+            return super().__new__(KojiNvr)
+        # If we get here, the prefix is not supported
+        raise ValueError(
+            f"Unsupported artifact ID format: '{raw_provider_id}'. "
+            f"Supported formats are: 'koji.build:', 'koji.task:', 'koji.nvr:'"
```

Comment:
```suggestion
            f"Supported formats are: 'koji.build:BUILD_ID', 'koji.task:TASK_ID', 'koji.nvr:NVR'"
```

Not sure if the format should not mention here the additional required data for all prefixes?

Reply:
Open to suggestions :)

Reply:
ok, if won't be used, this resolving